packages/react-components/src/atoms/Dropdown.tsx

```diff
@@ -1,51 +1,143 @@
-import { FC } from 'react';
-import Select, { OptionTypeBase } from 'react-select';
+import {
+  ComponentProps,
+  createContext,
+  createRef,
+  FC,
+  useContext,
+  useState,
+} from 'react';
+import Select, { OptionTypeBase, components } from 'react-select';
 import { css } from '@emotion/react';
+import { v4 as uuidV4 } from 'uuid';
 
 import { noop } from '../utils';
-import { validationMessageStyles } from '../form';
 import { dropdownChevronIcon } from '../icons';
 import { Option, reactSelectStyles } from '../select';
+import { useValidation, validationMessageStyles } from '../form';
 
 const containerStyles = css({
   flexBasis: '100%',
 });
 const DropdownIndicator: FC = () => dropdownChevronIcon;
 
+const InputContext = createContext<
+  ReturnType<typeof useValidation>['validationTargetProps'] &
+    Pick<DropdownProps<string>, 'required'>
+>({
+  ref: createRef(),
+  onBlur: noop,
+  onInvalid: noop,
+});
+
+const Input: React.FC<ComponentProps<typeof components.Input>> = (props) => {
+  const { ref, onBlur, ...inputProps } = useContext(InputContext);
+
+  return (
+    <components.Input
+      {...props}
+      {...inputProps}
+      onBlur={(event) => {
+        props.onBlur?.(event);
+        onBlur(event);
```

Comment:
I think we are triggering this twice as much as we need. handleInputValidation triggers this onBlur.

Reply:
yeah, `handleInputValidation` should trigger `onInvalid`